test/check-size.sh

```diff
@@ -0,0 +1,34 @@
+# A shell script to check if the binary size
+# has increased beyond a certain threshold
+# across commits. Meant to be used in CI with
+# git rebase <base branch>^ -x check-size.sh
+
+set -e
+
+BIN_NAME=checkpointctl
+PREV_SIZE_FILE=prev_size
+# Maximum allowable size difference, in bytes
+MAX_DIFF=51200
+
+# Build the checkpointctl binary
+make
+# Store the binary size
+BIN_SIZE=$(stat -c%s "$BIN_NAME")
+
+if [[ -f "$PREV_SIZE_FILE" ]]; then
+	# Read the previous size from the file
+	PREV_SIZE=$(cat "$PREV_SIZE_FILE")
+	# Calculate the difference between current and previous size
+	DIFF=$((BIN_SIZE - PREV_SIZE))
+	if [[ $DIFF -gt $MAX_DIFF ]]; then
+		echo "FAIL: size difference of \"$DIFF\" B exceeds limit \"$MAX_DIFF\" B"
+		exit 1
+	else
+		echo "PASS: size difference of \"$DIFF\" B within limit \"$MAX_DIFF\" B"
+	fi
+else
+	echo "No previous size present, storing current size"
```

Comment:
Here it might be good to add a comment like "_First time through: preserve original file size_",
similar to the [make-and-check-size](https://github.com/containers/podman/blob/aee7a3c16aee82d11689db91e42f82e52a690d98/hack/make-and-check-size#LL132C9-L132C58) script.